apps/remote_control/lib/lexical/remote_control/code_intelligence/definition.ex

```diff
@@ -1,17 +1,40 @@
 defmodule Lexical.RemoteControl.CodeIntelligence.Definition do
   alias Future.Code, as: Code
   alias Lexical.Ast
+  alias Lexical.Ast.Analysis
   alias Lexical.Document
   alias Lexical.Document.Location
   alias Lexical.Document.Position
+  alias Lexical.Formats
   alias Lexical.RemoteControl.CodeIntelligence.Entity
+  alias Lexical.RemoteControl.Search.Store
   alias Lexical.Text
 
   def definition(%Document{} = document, %Position{} = position) do
-    document
+    with {:ok, _, analysis} <- Document.Store.fetch(document.uri, :analysis),
+         {:ok, entity, _range} <- Entity.resolve(analysis, position) do
+      fetch_definition(entity, analysis, position)
+    end
+  end
+
+  defp fetch_definition({type, entity}, %Analysis{} = _analysis, %Position{} = _position)
+       when type in [:struct, :module] do
+    module = Formats.module(entity)
+
+    with [entry] <- Store.exact(module, type: type, subtype: :definition),
+         {:ok, location} <- to_location(entry) do
+      {:ok, location}
+    else
+      _ ->
+        {:ok, nil}
```

Comment:
Shouldn't this be :error?

Reply:
Only if the index doesn't take effect will it reach here, though an error also makes sense.

Changed.